### PR TITLE
Update CONTRIBUTING.md: publication instructions

### PR DESCRIPTION
**Pull request by konrad-jamrozik**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -161,7 +160,6 @@ npm list -g oad # Should denote no packages installed
 [the cache]: https://docs.npmjs.com/cli/v10/configuring-npm/folders#cache
 [`openapi-diff-build`]: https://dev.azure.com/azure-sdk/internal/_build/index?definitionId=7030
 [`public.openapi-diff`]: https://dev.azure.com/azure-sdk/public/_build?definitionId=135&_a=summary
-[upstream feeds of `openapi-platform`]: https://devdiv.visualstudio.com/DevDiv/_artifacts/feed/openapi-platform/Npm/@azure%2Foad/upstreams/
```
Comment (konrad-jamrozik) [inline CONTRIBUTING.md:164]: @scbedd I think you removed this as part of the pipeline migration work, yes?

Comment (konrad-jamrozik) [reply CONTRIBUTING.md:164]: Lol, sorry, now it loads. I was getting 404.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -143,26 +143,28 @@ npm list -g oad # Should denote no packages installed
...
-[`openapi-diff`]: https://github.com/Azure/openapi-diff
-[`dotnet restore`]: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-restore
+[#537791]: https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/537791
+[`@azure/oad` versions]: https://www.npmjs.com/package/@azure/oad?activeTab=versions
 [`dotnet build`]: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build
+[`dotnet restore`]: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-restore
 [`dotnet test`]: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-test
 [`npm link`]: https://docs.npmjs.com/cli/v10/commands/npm-link
-[the cache]: https://docs.npmjs.com/cli/v10/configuring-npm/folders#cache
 [`openapi-diff-build`]: https://dev.azure.com/azure-sdk/internal/_build/index?definitionId=7030
```
Comment (konrad-jamrozik) [inline CONTRIBUTING.md:164]: @mikeharder @scbedd these two:

[`openapi-diff-build`]: https://dev.azure.com/azure-sdk/internal/_build/index?definitionId=7030
[`public.openapi-diff`]: https://dev.azure.com/azure-sdk/public/_build?definitionId=135&_a=summary

Now appear to be based on the same build definition. Is there any reason to keep the public one around or should I delete it?

Comment (scbedd) [reply CONTRIBUTING.md:164]: IMO keep both use `internal` for release and nightly (if you care about that for this project) and the `public` build for your CI checks.

**Review by mikeharder — APPROVED** [review]

**Pull request merged**